stdlib/2/__builtin__.pyi

```diff
@@ -28,6 +28,7 @@ _TT = TypeVar('_TT', bound='type')
 class object:
     __doc__ = ...  # type: Optional[str]
     __class__ = ...  # type: type
+    __dict__ = ...  # type: Union[Dict[str, Any], Dict[unicode, Any]]
```

Comment:
I don't think a Union works well here. The most common use case is probably something like `obj.__dict__[some_str] = some_value`, but with this code mypy will complain about that because `__dict__` may have unicode keys. I don't think you can even fix that with `isinstance`.

I can't really think of a better solution though. Maybe we should pick one of unicode and str and accept that some code may not work.